pkg/manager/risingwave_controller_manager_impl.go

```diff
@@ -1175,7 +1176,16 @@ func (mgr *risingWaveControllerManagerImpl) syncObject(ctx context.Context, obj
 		newObj.SetResourceVersion(obj.GetResourceVersion())
 		logger.Info(fmt.Sprintf("Update the object of %s", gvk.Kind), "object", utils.GetNamespacedName(newObj),
 			"generation", mgr.risingwaveManager.RisingWave().Generation)
-		return mgr.client.Update(ctx, newObj)
+		if err = mgr.client.Update(ctx, newObj); err == nil {
+			return nil
+		}
+		if !mgr.forceUpdateEnabled ||
+			obj.GetLabels()[consts.LabelRisingWaveOperatorVersion] == newObj.GetLabels()[consts.LabelRisingWaveOperatorVersion] {
+			return err
+		}
+		if err := mgr.client.Delete(ctx, obj); err != nil {
+			return err
+		}
```

Comment:
I think we should only do `delete + create` when the error is a `Forbidden`. If the update succeeds, then everything's good.